src/c++/perf_analyzer/test_report_writer.cc

```diff
@@ -27,16 +27,29 @@
 #include <string>
 
 #include "doctest.h"
+#include "profile_data_collector.h"
 #include "report_writer.h"
 
 namespace triton { namespace perfanalyzer {
 
 class TestReportWriter : ReportWriter {
  public:
+  TestReportWriter() = default;
+  TestReportWriter(const std::shared_ptr<ProfileDataCollector>& collector)
```

Comment:
Not gonna block the review on this, but I think this would be good for a future story to shift towards the mock pattern we have elsewhere in the code:

https://github.com/triton-inference-server/client/blob/694cfd4c55f06b5ed2a3255794fe0c65057b5b81/src/c%2B%2B/perf_analyzer/test_infer_context.cc#L1

Basically, instead of needing to make a custom constructor for the class to set this private data member, the mock class can be made a friend of the superclass, then the mock class can define a public reference to the private superclass data member. I think it's cleaner than messing around with custom constructors. My opinion is that custom constructors for tests don't scale well. Imagine different tests need different things to happen to different data members (or even the same data members). That will be one constructor per test. That bloats the mock class code space when that logic should really live in the test case. That's where exposing public mock references to private superclass (the class being tested) is preferable. The test cases then have free access to the superclass private data members for manipulation and testing purposes.

Reply:
I agree. Mock pattern is potentially more cleaner way to have access to all the private members, but I decided not to go for it this time. ReportWriter is very small class and the changes it needs for the tests to work were only few lines, and I didn't wanted to introduce a mock header just to have a public reference to the `collector_`.